java-checks/src/main/java/org/sonar/java/checks/SwitchLastCaseIsDefaultCheck.java

```diff
@@ -42,35 +46,57 @@
 
   @Override
   public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
     SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
-    CaseLabelTree defaultLabel = getDefaultLabel(switchStatementTree);
-    CaseLabelTree lastLabel = getLastLabel(switchStatementTree);
-    if (defaultLabel == null) {
-      reportIssue(switchStatementTree.switchKeyword(), "Add a default case to this switch.");
-    } else if (!defaultLabel.equals(lastLabel)) {
-      reportIssue(defaultLabel, "Move this default to the end of the switch.");
+    Optional<CaseLabelTree> defaultLabel = getDefaultLabel(switchStatementTree);
+    if (!defaultLabel.isPresent()) {
```

Comment:
This could be reworked to avoid having ! in  nearly every condition to make the different cases cleaner to read.